### PR TITLE
[WFLY-10886] improve performance from AbstractJMSContext.getContext

### PR DESCRIPTION
**Pull request by fbrandaouolinc**
Change AbstractJMSContext.getContext from synchronized method to ConcurrentHashMap for improved performance

https://issues.redhat.com/browse/WFLY-10886

Upstream PR: https://github.com/wildfly/wildfly/pull/15077

### CODE REVIEW

**Review by saulocn — APPROVED** [review]

File: messaging-activemq/injection/src/test/java/org/wildfly/extension/messaging/activemq/deployment/injection/AbstractJMSContextTestCase.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
+package org.wildfly.extension.messaging.activemq.deployment.injection;
```
Comment (darranl) [inline messaging-activemq/injection/src/test/java/org/wildfly/extension/messaging/activemq/deployment/injection/AbstractJMSContextTestCase.java:1]: All new classes should have a copyright header added with the year set to the current year,

**Comment by darranl** [comment]
@ehsavoie Could you please give this a review?

**Review by felipe-toshio — APPROVED** [review]

**Comment by ehsavoie** [comment]
Could you please squash your commits ?
Otherwise looks good to me.

**Review by ehsavoie — CHANGES_REQUESTED** [review]
squash the commits please

**Comment by fbrandaouolinc** [comment]
Hi @ehsavoie , @darranl ,

Do you have any other considerations?

Thanks

**Comment by fbrandaouolinc** [comment]
@jmesnil can you see this PR, please?

**Review by ehsavoie — APPROVED** [review]

**Comment by darranl** [comment]
Sorry just taken off "ready for merge" do we also have a PR against main?

**Comment by fbrandaouolinc** [comment]
> Sorry just taken off "ready for merge" do we also have a PR against main?

No, we do not have a PR against main.
I'll open, okay?

**Comment by fbrandaouolinc** [comment]
@darranl I've create the PR against main: https://github.com/wildfly/wildfly/pull/15077

**Comment by bstansberry** [comment]
I just kicked off another run of the "Linux with security manager" job, but reconfigured to use -Dts.multinode instead of -DallTests, which means most of the integration testsuite won't run, but the module where the current job had failures will. So it will take much less time to run. I'd be shocked if this PR has anything to do with the current failures.

I'm not concerned about the 1 failure in "Linux - JDK 8" as that tests is well known for intermittent failures.

**Comment by fbrandaouolinc** [comment]
> I just kicked off another run of the "Linux with security manager" job, but reconfigured to use -Dts.multinode instead of -DallTests, which means most of the integration testsuite won't run, but the module where the current job had failures will. So it will take much less time to run. I'd be shocked if this PR has anything to do with the current failures.
> 
> I'm not concerned about the 1 failure in "Linux - JDK 8" as that tests is well known for intermittent failures.

Hi, the PR against main have failures in differents tests of this 2 suites.

**Comment by bstansberry** [comment]
Thanks @fbrandaouolinc

**Pull request merged**